pkg/splunk/enterprise/util.go

```diff
@@ -64,10 +64,21 @@ func ApplySplunkConfig(client splcommon.ControllerClient, cr splcommon.MetaObjec
 	}
 
 	// create or retrieve splunk secrets
-	secrets := getSplunkSecrets(cr, instanceType, idxcSecret, pass4SymmKey)
-	secrets.SetOwnerReferences(append(secrets.GetOwnerReferences(), splcommon.AsOwner(cr)))
-	if secrets, err = splctrl.ApplySecret(client, secrets); err != nil {
-		return nil, err
+	var secrets *corev1.Secret
+	if instanceType == SplunkIndexer && len(spec.IndexerClusterRef.Name) > 0 {
+		secrets = &corev1.Secret{}
+		secretRefName := GetSplunkSecretsName(spec.IndexerClusterRef.Name, SplunkIndexer)
+		namespacedName := types.NamespacedName{Namespace: cr.GetNamespace(), Name: secretRefName}
```

Comment:
Anywhere we reference IndexerClusterRef, we may want to allow that to also specify namespace (which is not the same). at least in other cases for all the refs this is the pattern I tried to follow.